src/lib/pickles/test/ec-scaling-foreign/test.ml

```diff
@@ -0,0 +1,348 @@
+open Core_kernel
+open Pickles.Impls.Step
+
+module type S = sig
+  type foreign
+
+  val foreign_typ : (foreign, unit) Typ.t
+
+  val zero : foreign
+
+  val assert_equal : foreign -> foreign -> unit
+
+  val range_check : foreign -> unit
+
+  val add_chain : range_check:(foreign -> unit) -> foreign -> foreign -> foreign
+
+  val sub_chain : range_check:(foreign -> unit) -> foreign -> foreign -> foreign
+
+  val mul_chain : range_check:(foreign -> unit) -> foreign -> foreign -> foreign
+
+  val add : range_check:(foreign -> unit) -> foreign -> foreign -> foreign
+
+  val mul : range_check:(foreign -> unit) -> foreign -> foreign -> foreign
+end
+
+module Make (Foreign_implementation : S) = struct
+  include Foreign_implementation
+
+  (*
+   x = s^2 - A_x - B_x
+   y = s * (B_x - x) - B_y
+
+   s = (A_y - B_y) / (A_x - B_x)
+*)
+  let ec_add (a_x, a_y) (b_x, b_y) =
+    let actual_range_check = range_check in
+    let range_checks = ref [] in
+    let range_check x = range_checks := x :: !range_checks in
+    (* diff_x = a_x - b_x *)
+    let diff_x = sub_chain ~range_check a_x b_x in
+    let s = exists foreign_typ ~compute:(fun () -> ()) in
+    (* diff_y = diff_x * s *)
+    let diff_y = mul_chain ~range_check diff_x s in
+    (* diff_y = a_y - b_y *)
+    let a_y' = add_chain ~range_check diff_y b_y in
+    assert_equal a_y a_y' ;
+    (* s_squared = s * s *)
+    let s_squared = mul_chain ~range_check s s in
+    (* s_squared_sub_a_x = s * s - a_x *)
+    let s_squared_sub_a_x = sub_chain ~range_check s_squared a_x in
+    (* x = s * s - a_x - b_x *)
+    let x = sub_chain ~range_check s_squared_sub_a_x b_x in
+    (* x_change = b_x - x *)
+    let neg_x_change = sub_chain ~range_check x b_x in
+    (* y_change = (b_x - x) * s *)
+    let neg_y_change = mul_chain ~range_check neg_x_change s in
+    let y = exists foreign_typ ~compute:(fun () -> ()) in
```

Comment:
```suggestion
    let y = exists foreign_typ ~compute:Fn.id in
```

Reply:
Consider applying the suggestion or making a definition like `let id_unit () = ()` at the top of this file for this function that is used multiple times.

Reply:
No